lib/collection/response.js

```diff
@@ -142,17 +142,24 @@ var util = require('../util'),
     supportsBuffer = (typeof Buffer !== undefined) && _.isFunction(Buffer.byteLength),
 
     /**
-     * Normalises an input Buffer or buffer.toJSON() into a Buffer or ArrayBuffer.
+     * Normalizes an input Buffer, Buffer.toJSON() or base64 string into a Buffer or ArrayBuffer.
      *
      * @private
-     * @param {Buffer|Object} stream - An instance of Buffer of an object representation of Buffer(Buffer.toJSON())
+     * @param {Buffer|Object|String} stream - An instance of Buffer, Buffer.toJSON(), or Buffer.toString('base64')
      * @returns {Buffer|ArrayBuffer|undefined}
      */
-    normaliseStream = function (stream) {
+    normalizeStream = function (stream) {
+        // create buffer from buffer's JSON representation
         if (stream && stream.type === BUFFER && _.isArray(stream.data)) {
             // @todo Add tests for Browser environments, where ArrayBuffer is returned instead of Buffer
-            return typeof Buffer === FUNCTION ? new Buffer(stream.data) : new Uint8Array(stream.data).buffer;
+            return typeof Buffer === FUNCTION ? Buffer.from(stream.data) : new Uint8Array(stream.data).buffer;
         }
+        // create buffer from base64 string
+        else if (typeof stream === 'string') {
+            return Buffer.from(stream, 'base64');
```

Comment:
There should be an identification saying the stream is base64 somehow before we default decode. Reason being that otherwise the default would prevent us from modifying stream structure and yet retain compatibility with older SDK running in older apps.

Reply:
Worst case there should be a way to indicate to not process the stream in case we choose any other format in future.